### PR TITLE
CLN: Numpy compat functions namespace

### PR DESCRIPTION
**Pull request by phofl**
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

Should we do something about the private variables ``_np_version_under1p19`` and ``_np_version_under1p20`` too?

cc @jreback

### CODE REVIEW

**Comment by jreback** [comment]
let's save this until we revert the numpy fixes

**Comment by phofl** [comment]
Sounds good, will fix the errors in the meantime. Will reopen when reverted

**Comment by phofl** [comment]
Reopening since the other commit was reverted. Should we make ``_np_version_under1p19`` and ``_np_version_under1p20`` make non private to move them too?

File: pandas/tests/io/parser/test_parse_dates.py [hunk 1]
```diff
@@ -15,8 +15,7 @@
 from pandas._libs.tslib import Timestamp
 from pandas._libs.tslibs import parsing
 from pandas._libs.tslibs.parsing import parse_datetime_string
-from pandas.compat import is_platform_windows
-from pandas.compat.numpy import np_array_datetime64_compat
+from pandas.compat import is_platform_windows, np_array_datetime64_compat
```
Comment (jreback) [inline pandas/tests/io/parser/test_parse_dates.py:18]: OT: we can prob remove this np_array_datetime64_compat it is pretty old stuff

File: pandas/tests/scalar/period/test_period.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 from pandas._libs.tslibs.parsing import DateParseError
 from pandas._libs.tslibs.period import INVALID_FREQ_ERR_MSG, IncompatibleFrequency
 from pandas._libs.tslibs.timezones import dateutil_gettz, maybe_get_tz
-from pandas.compat.numpy import np_datetime64_compat
+from pandas.compat import np_datetime64_compat
```
Comment (jreback) [inline pandas/tests/scalar/period/test_period.py:13]: this too (but just make a new issue for now)

**Comment by jreback** [comment]
> Reopening since the other commit was reverted. Should we make `_np_version_under1p19` and `_np_version_under1p20` make non private to move them too?

yes

**Comment by phofl** [comment]
Done, opened #39283

**Comment by jreback** [comment]
thanks @phofl

**Pull request merged**